brave/brave-core
# Add current_version flag to P3A JSON message

Resolves brave/brave-browser#21622

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

Must use MITM proxy to validate P3A messages being sent.

To test main mode (uses usage ping info):

0. Install slightly older browser version (which includes this change) for a given channel, ensure usage ping is enabled
1. Launch browser, wait for usage ping to be sent, close browser
2. Advance date by 10 days, launch browser, wait for P3A messages to be sent.
3. Ensure that request payloads sent to p3a-json.brave.com contain `"current_version": false`.
4. Advance date by 25 days, launch browser, wait for P3A messages to be sent.
5. Ensure that request payloads sent to p3a-json.brave.com still contain `"current_version": false`.
6. Upgrade to latest browser version for given channel, advance time by one day.
7. Launch browser, wait for usage ping to be sent, close browser
8. Advance date by 10 days, launch browser, wait for P3A messages to be sent.
9. Ensure that request payloads sent to p3a-json.brave.com contain `"current_version": true`.
10. Advance date by 25 days, launch browser, wait for P3A messages to be sent.
11. Ensure that request payloads sent to p3a-json.brave.com still contain `"current_version": true`.

To test fallback mode:

0. Install slightly older browser version (which includes this change) for a given channel
1. Launch browser, disable usage ping, close browser
2. Advance date by 10 days, launch browser, wait for P3A messages to be sent.
4. Ensure that request payloads sent to p3a-json.brave.com contain `"current_version": true`.
5. Advance date by 25 days, launch browser, wait for P3A messages to be sent.
6. Ensure that request payloads sent to p3a-json.brave.com contain `"current_version": false`.
7. Upgrade to latest browser version for given channel.
8. Advance date by 10 days, launch browser, wait for P3A messages to be sent.
9. Ensure that request payloads sent to p3a-json.brave.com contain `"current_version": true`.
11. Advance date by 25 days, launch browser, wait for P3A messages to be sent.
12. Ensure that request payloads sent to p3a-json.brave.com contain `"current_version": false`.
